### PR TITLE
Add result section header

### PR DESCRIPTION
**Pull request by mathewscullin**
## Overview

Added results header to `SearchTableViewController`.

## Changes Made

Added `SearchTableViewHeader` and respective functions in `SearchTableViewController`.

## Test Coverage

I ran the simulator to ensure the header updates with the correct number of sections returned in the query.

## Related PRs or Issues (delete if not applicable)

#32 

## Screenshots (delete if not applicable)

<details>

  <summary>Some Results</summary>

  
  
Simulator Screen Shot - iPhone 11 Pro - 2020-02-19 at 19 25 19

</details>

<details>

  <summary>No Results</summary>

  
  
Simulator Screen Shot - iPhone 11 Pro - 2020-02-19 at 19 25 55

</details>

### CODE REVIEW

File: CourseGrab/ViewControllers/Search/SearchTableViewHeader.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import UIKit
+
+class SearchTableViewHeader: UITableViewHeaderFooterView {
+
+    private let titleLabel = UILabel()
+
+    override init(reuseIdentifier: String?) {
+        super.init(reuseIdentifier: reuseIdentifier)
+
+        contentView.backgroundColor = .white
```
Comment (ReadeP) [inline CourseGrab/ViewControllers/Search/SearchTableViewHeader.swift]: newline

File: CourseGrab/ViewControllers/Search/SearchTableViewHeader.swift [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        titleLabel.font = ._20Semibold
+        titleLabel.textColor = .black
+        contentView.addSubview(titleLabel)
+        
+        titleLabel.snp.remakeConstraints { make in
```
Comment (ReadeP) [inline CourseGrab/ViewControllers/Search/SearchTableViewHeader.swift]: Why `remakeConstraints` and not `makeConstraints`?

File: CourseGrab/ViewControllers/Search/SearchTableViewController.swift [hunk 3]
```diff
@@ -74,11 +66,17 @@ extension SearchTableViewController {
         cell.configure(for: sections[indexPath.row])
         return cell
     }
-
+
```
Comment (kevinchan159) [inline CourseGrab/ViewControllers/Search/SearchTableViewController.swift]: remove whitespace

Comment (mathewscullin) [reply CourseGrab/ViewControllers/Search/SearchTableViewController.swift]: In the file there is only one line of whitespace, that is protocol between functions right? @kevinchan159

**Pull request merged**